### ISSUE TITLE
Incorrect handling of `__x` in typeshed hints

### ISSUE THREAD

**Issue by penn5** [microsoft/pyright#4619]
**Describe the bug**
When writing a protocol to describe an operator described by a typeshed function that accepts positional-only parameters, the protocol must accept parameters with names matching the names specified in the typeshed.

**To Reproduce**
The code below does not pass pyright if the `__x` parameters are made positional or renamed.

**Expected behavior**
The code should pass regardless of the naming of parameters, as the actual signature of `int.__add__` is `(/, int) -> int`.

**Screenshots or Code**
```python
import time
import typing
I = typing.TypeVar("I", contravariant=True)
O = typing.TypeVar("O", covariant=True)

class LAddable(typing.Protocol[I, O]):
    def __add__(self, __x: I) -> O:
        ...

class RAddable(typing.Protocol[I, O]):
    def __radd__(self, __x: I) -> O:
        ...

@typing.overload
def slow_add(a: I, b: RAddable[I, O]) -> O:
    ...

@typing.overload
def slow_add(a: LAddable[I, O], b: I) -> O:
    ...

def slow_add(a, b):  # type: ignore
    time.sleep(0.1)
    return a + b

if __name__ == "__main__":
    slow_add(1, 2)
```

**VS Code extension or command-line**
pyright 1.1.293

**Additional context**
https://www.reddit.com/r/Python/comments/10zdidm/why_type_hinting_sucks/

**Comment by erictraut** [comment]
I'm not able to repro the problem you're reporting. Can you verify that you are seeing an error with the above code?

**Comment by penn5** [comment]
As noted, the code I provided works. The issue only occurs if the names of the `__add__` and `__radd__`'s parameters are changed. 

For example:
```python
class LAddable(typing.Protocol[I, O]):
    def __add__(self, x: I) -> O:
        ...

class RAddable(typing.Protocol[I, O]):
    def __radd__(self, x: I) -> O:
        ...
```

```
No configuration file found.
No pyproject.toml file found.
stubPath /home/penn/Code/PycharmProjects/pythonProject5/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.293
0 errors, 0 warnings, 0 informations 
Completed in 0.875sec
(venv) [penn@Ferao pythonProject5]$ pyright /home/penn/.config/JetBrains/PyCharm2022.3/scratches/scratch_7.py
No configuration file found.
No pyproject.toml file found.
stubPath /home/penn/Code/PycharmProjects/pythonProject5/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.293
~/.config/JetBrains/PyCharm2022.3/scratches/scratch_7.py
  ~/.config/JetBrains/PyCharm2022.3/scratches/scratch_7.py:27:14 - error: Argument of type "Literal[1]" cannot be assigned to parameter "a" of type "LAddable[I@slow_add, O@slow_add]" in function "slow_add"
    "Literal[1]" is incompatible with protocol "LAddable[int, O@slow_add]"
      "__add__" is an incompatible type
        Type "(__x: int, /) -> int" cannot be assigned to type "(x: I@LAddable) -> O@LAddable"
          Position-only parameter mismatch; expected 1 but received 0 (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 0.809sec
```

**Comment by erictraut** [comment]
Yes, this is correct. If you change the protocol classes `LAddable` and `RAddable` to indicate that compatible types must provide methods that are callable with keyword arguments, then instances of `int` will not be compatible with the protocol. The methods `int.__add__` and `int.__radd__` cannot be called with a keyword argument `x`.

**Comment by penn5** [comment]
That makes sense. I encountered the same issue using the `/` syntax to denote positional-only arguments (for `x`)
